hal/include/HAL/Ports.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <stdint.h>
+
+namespace hal {
+constexpr int kNumAccumulators = 2;
+constexpr int kNumAnalogTriggers = 8;
+constexpr int kNumAnalogInputs = 8;
+constexpr int kNumAnalogOutputs = 2;
+constexpr int kNumCounters = 8;
```

Comment:
It feels like these enums shouldn't be part of the public headers; putting them here encourages users to get these values directly instead of via the API calls, and prevents us from setting them to the actual hardware values from the FPGA (e.g. tCounter::kNumSystems) so they're guaranteed to be kept in sync.